### PR TITLE
Fix a nil pointer issue

### PR DESCRIPTION
**Pull request by soothing-rain**
/kind bug

issue: #18270

Signed-off-by: Yuchen Gao <yuchen.gao@zilliz.com>

### CODE REVIEW

File: internal/datacoord/meta.go [hunk 1]
```diff
@@ -800,11 +800,11 @@ func (m *meta) CompleteMergeCompaction(compactionLogs []*datapb.CompactionSegmen
 
 	var startPosition, dmlPosition *internalpb.MsgPosition
 	for _, s := range segments {
-		if dmlPosition == nil || s.GetDmlPosition().Timestamp < dmlPosition.Timestamp {
+		if dmlPosition == nil || s.GetDmlPosition().GetTimestamp() < dmlPosition.GetTimestamp() {
 			dmlPosition = s.GetDmlPosition()
 		}
 
-		if startPosition == nil || s.GetStartPosition().Timestamp < startPosition.Timestamp {
+		if startPosition == nil || s.GetStartPosition().GetTimestamp() < startPosition.GetTimestamp() {
```
Comment (congqixia) [inline internal/datacoord/meta.go:807]: maybe we shall check whether `s.GetStartPostion()` and `s.GetDmlPosition()` is nil

Comment (soothing-rain) [reply internal/datacoord/meta.go:807]: That should be OK, 0 will be returned in `s.GetStartPostion()` or `s.GetDmlPosition()` is nil:

```
func (m *MsgPosition) GetTimestamp() uint64 {
	if m != nil {
		return m.Timestamp
	}
	return 0
}
```

Comment (congqixia) [reply internal/datacoord/meta.go:807]: If the segment startPosition or dmlPosition is nil, using GetStartPosition().GetTimeStamp() shall always less or equal to current minimal postion, which means a nil value will overwrite the minimal position.

Comment (soothing-rain) [reply internal/datacoord/meta.go:807]: Sure that makes sense.

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**